### ISSUE TITLE
[QA] “Error: missing revert data in call exception” is displayed in Console

### ISSUE THREAD

**Issue by tsadovska** [wormhole-foundation/wormhole-connect#228]
#### Steps:
1. Connect Metamask wallet in ‘From’ section via Wallet Connect > QR Code
2. Connect Solflare wallet in ‘To’ section
3. Prepare to send 0.012 ETH from Goerli to Solana
4. Click on “Approve and proceed with transaction” button several times
 
#### Actual result:
Error is displayed in Console:

```
Send.tsx:106 Error: missing revert data in call exception; Transaction reverted without a reason string [ See: https://links.ethers.org/v5-errors-CALL_EXCEPTION ] (data="0x", transaction={"from":"0x0a370E7c5F7F944d451B48bc151a7BEC454bc185","gasLimit":{"type":"BigNumber","hex":"0x03d090"},"to":"0xF890982f9310df57d00f659cf4fd87e65adEd8d7","value":{"type":"BigNumber","hex":"0x2aa1efb94e0000"},"data":"0x9981509f00000000000000000000000000000000000000000000000000000000000000019c6746b429022c874c3679e8147dda94890c313a1a74325f976863de7e8e9cac000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000c3110100","accessList":null}, error={"code":-32000,"message":"err: insufficient funds for gas * price + value: address 0x0a370E7c5F7F944d451B48bc151a7BEC454bc185 have 0 want 12000000000000000 (supplied gas 250000)"}, code=CALL_EXCEPTION, version=providers/5.7.2)
    at Logger.makeError (index.ts:269:1)
    at Logger.throwError (index.ts:281:1)
    at checkError (json-rpc-provider.ts:66:1)
    at Web3Provider.<anonymous> (json-rpc-provider.ts:642:1)
    at Generator.throw (<anonymous>)
    at rejected (json-rpc-provider.ts:1:1)
```

#### Expected result:
Transaction is processed without errors.

#### Tested devices:
MacBook Pro, macOS Monterey 12.6.1, Chrome Version 111.0.5563.64 (Official Build) (x86_64)
Samsung S20 (Android 12), Chrome Version: 110.0.5481.63

#### Attachment:

![Screenshot 2023-03-20 at 14 49 05](https://user-images.githubusercontent.com/10746654/226344236-46d42e4a-f247-4bdd-aec5-7159d7677d53.png)

**Comment by tsadovska** [comment]
Issue is reproducible with the similar scenario on https://sui-support--wormhole-connect.netlify.app/ 

**Steps:**

1. Connect Sui wallet in ‘From’ section
2. Connect Metamask wallet via WalletConnect in ’To’ section
3. Send 0.01 WBNB from Sui to BSC
4. Click “Claim” button

**Actual results:**

- Error “Your claim has failed, please try again” is displayed for the user
- “Error: missing revert data in call exception” is displayed in Console

**Expected results:**
Transaction is processed without errors.

**Attachments:**
![Screenshot 2023-05-03 at 15 52 49](https://user-images.githubusercontent.com/10746654/235922563-80f78431-497c-40ff-8116-73051336d855.png)


https://user-images.githubusercontent.com/10746654/235922601-9dd15220-fb4e-4a12-b1a1-fd99850d3c88.mov

**Comment by AigaD** [comment]
Same for transaction Sui - Fuji

Steps:
1. Prepare to send 0.002 WAVAX from Sui to Fuji
2. Click on warning message - manual Claim
3. Choose pay with two transactions
4. Click Approve and proceed with transaction
5. When possible click Claim

Attachment:

https://user-images.githubusercontent.com/91047061/235924567-e5cb88c1-9594-46d2-9a8d-a5da780e99cd.mov

**Comment by AigaD** [comment]
Funds were deducted from SUI wallet, but where not received on Metamask wallet.
Tested from: https://mainnet--wormhole-connect.netlify.app/

Attachments:
<img width="1025" alt="Screenshot 2023-05-05 at 08 39 08" src="https://user-images.githubusercontent.com/91047061/236384554-4c1395c8-e4c3-4f1c-96a3-80eb0a0397d9.png">
<img width="897" alt="Screenshot 2023-05-05 at 08 38 54" src="https://user-images.githubusercontent.com/91047061/236384559-dc826cf0-75ba-4755-bb3c-b39932fc5584.png">

<img width="1332" alt="Screenshot 2023-05-05 at 08 41 11" src="https://user-images.githubusercontent.com/91047061/236384662-fc00b9a3-29c3-43d5-9067-1f6385c7417d.png">

**Comment by tsadovska** [comment]
Issue is also reproducible on https://mainnet--wormhole-connect.netlify.app/ with the following scenario:

1. Connect DeFi wallet through WalletConnect in ‘From’ section
2. Connect Sui wallet in ’To’ section
3. Prepare to send 0.00155 BNB from BSC to Sui
4. Click “Approve and proceed with transaction” button

Attachment:
![Screenshot 2023-05-05 at 11 29 40](https://user-images.githubusercontent.com/10746654/236411909-32ea7906-b68a-489a-8d23-4b4811017d56.png)

**Comment by tsadovska** [comment]
Issue is reproducible on https://feature-sei--wormhole-connect.netlify.app/ [Testnet] with the following scenario:
1. Connect "Coin98 for SEI" wallet in 'From' section
2. Connect Sei network in 'From' section
3. Connect Metamask wallet through WalletConnect in 'To' section
4. Connect Goerli network in 'To' section
5. Select SEI asset
6. Type Amount, in my case 0.001
7. Click "Approve and proceed with transaction" button
8. Click "Claim" button on Redeem screen

Actual result: Loading does not finish, [Sei Explorer](https://sei.explorers.guru/transaction/B01A953EB9C96C93055920913597869C3F222B386662DBD6A71F08A7ACC89C4B). Error is displayed in Console:
```
Uncaught (in promise) Error: missing revert data in call exception; Transaction reverted without a reason string [ See: https://links.ethers.org/v5-errors-CALL_EXCEPTION ] (data="0x", transaction={"from":"0x0a370E7c5F7F944d451B48bc151a7BEC454bc185","to":"0xF890982f9310df57d00f659cf4fd87e65adEd8d7","data":"0xaa4efa5badb25765401286ff8111d1a8ce49304d0bde60e106f26dfc6820f7519eac6072","accessList":null}, error={}, code=CALL_EXCEPTION, version=providers/5.7.2)
    at f.makeError (index.ts:269:28)
    at f.throwError (index.ts:281:20)
    at y (json-rpc-provider.ts:66:16)
    at u.<anonymous> (json-rpc-provider.ts:642:20)
    at Generator.throw (<anonymous>)
    at s (formatter.ts:523:1)
```

![Screenshot 2023-06-19 at 12 28 49](https://github.com/wormhole-foundation/wormhole-connect/assets/10746654/3800a0bb-b75c-464b-9861-b71dc0052042)

**Comment by tsadovska** [comment]
Issue is reproducible on https://wormhole-connect.netlify.app/ with the following scenario:

#### Steps:
1. Connect Metamask wallet through Wallet Connect in From section
2. Connect Spika wallet in To section
3. Prepare to send 0.01 MATIC from Mumbai to Aptos
4. Click "Approve and proceed with transaction"

#### Actual Result:
* "Error sending transfer, please try again" is displayed for the user
* Error is displayed in Console:
```
Error: missing revert data in call exception; Transaction reverted without a reason string [ See: https://links.ethers.org/v5-errors-CALL_EXCEPTION ] (data="0x", transaction={"from":"0x0a370E7c5F7F944d451B48bc151a7BEC454bc185","to":"0x377D55a7928c046E18eEbb61977e714d2a76472a","value":{"type":"BigNumber","hex":"0x2386f26fc10000"},"data":"0x9981509f00000000000000000000000000000000000000000000000000000000000000164e5be942eb4587f43931940a5a929268613a7586e226bdd32b097484b5ee62e8000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000207a0000","accessList":null}, error={}, code=CALL_EXCEPTION, version=providers/5.7.2)
    at l.makeError (main.f2ca49db.js:37:1803301)
    at l.throwError (main.f2ca49db.js:37:1803421)
    at g (main.f2ca49db.js:37:1860317)
    at u.<anonymous> (main.f2ca49db.js:37:1870501)
    at Generator.throw (<anonymous>)
    at s (main.f2ca49db.js:37:1859648)
```

#### Expected result:
Transaction proceeds without errors.

#### Attachment:
![Screenshot 2023-06-24 at 13 36 00](https://github.com/wormhole-foundation/wormhole-connect/assets/10746654/b61a6a1a-5454-45da-ba20-bb511755ea9d)

**Comment by AigaD** [comment]
This has been reproduced for transaction Aptos APT -> Polygon and APT -> Ethereum
Tested from: https://mainnet--wormhole-connect.netlify.app/#
Attachment:

https://github.com/wormhole-foundation/wormhole-connect/assets/91047061/c77e8130-488d-4af4-8c7b-28efb7d71e25

<img width="1381" alt="Screenshot 2023-06-26 at 12 43 05" src="https://github.com/wormhole-foundation/wormhole-connect/assets/91047061/908de6ad-8628-4667-b0d7-1d31e5375d17">


<img width="1361" alt="Screenshot 2023-06-26 at 14 05 52" src="https://github.com/wormhole-foundation/wormhole-connect/assets/91047061/2bfe913e-213b-4310-a2cd-b5c07bd5a2cf">

**Comment by tsadovska** [comment]
Reviewed scenarios that are mentioned in the ticket.

Testnet:
* Send ETH from Goerli (Metamask) to Solana (Solflare) - will be rechecked after [#593](https://github.com/wormhole-foundation/wormhole-connect/issues/593) 
* Send ETH from Goerli (Metamask) to Solana (Phantom) - Tested OK ([Goerli](https://goerli.etherscan.io/tx/0xd2bfbaeca3e6583f5daa98cba91a261059df5d7ccb80413743fb753e2232ef3d), [Solana](https://explorer.solana.com/tx/5zYKExK1Q4WzaX9mY5aYafh1dEtSADztDeEJ7GoGtHfFPjJjpq3aL4eAB4rtwWiTMSRv2DBqfUQGws42QujsE5aT?cluster=devnet))
* Send SOL from Solana (Solflare) to Mumbai (Metamask through WalletConnect) - will be rechecked after [#747](https://github.com/wormhole-foundation/wormhole-connect/issues/747)
* Send WBNB from Sui (Sui) to BSC (Metamask) - will be rechecked after [#537](https://github.com/wormhole-foundation/wormhole-connect/issues/537), [#534](https://github.com/wormhole-foundation/wormhole-connect/issues/534) 
* Send WAVAX from Sui (Sui) to Fuji (Metamask) - will be rechecked after [#537](https://github.com/wormhole-foundation/wormhole-connect/issues/537), [#534](https://github.com/wormhole-foundation/wormhole-connect/issues/534) 
* Send SEI from Sei (Coin98 for SEI) to Goerli (Metamask through WalletConnect) - will be rechecked after [#697](https://github.com/wormhole-foundation/wormhole-connect/issues/697) 
* Send MATIC from Mumbai (Metamask through WalletConnect) to Aptos (Spika) - will be rechecked after [#747](https://github.com/wormhole-foundation/wormhole-connect/issues/747)
* Send MATIC from Mumbai (Metamask) to Aptos (Spika) - will be rechecked after [#764](https://github.com/wormhole-foundation/wormhole-connect/issues/764)

Mainnet:
* Send SUI from Sui (Sui) to Polygon (Metamask) - Tested NOK ([Sui](https://suiexplorer.com/txblock/5ZwympN7EhLT4gwEGx3VwNnpc5Sa7FNQq3VukZKo3nEj), claiming fails)  

Error is displayed in Console:
```
Error: missing revert data in call exception; Transaction reverted without a reason string [ See: https://links.ethers.org/v5-errors-CALL_EXCEPTION ] (data="0x", transaction={"from":"0x9631288F4050F7CFbf77B77f8540DeCF6cfC7012","gasLimit":{"type":"BigNumber","hex":"0x03d090"},"to":"0x5a58505a96D1dbf8dF91cB21B54419FC36e93fdE","data":"0xc68785190000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000041801000000030d00ae06ccffbc521121320810d6e82f470fa79b15a1d30fafc6f529af29cbe168bf5bff2165ee214fe754f44d64aa9a7535bd3b6c6387bc0c4728a70aba42ebe1b7000171952cfafcb3ce21ea0ed748e3b944fc12f9cfcff4bf444d23667a72514e429218358e76cfc548feeb6cc054a0de50bb8ca971fe17a465147d2af833158a733a010241856e885f1b8278e6c7e55c22667bae725f979bf9521d2d1defbdf19170905c5965e461990426db9d9d81e9ebba2dd06706b0e24e946f699aad08bfe3ed1ea701039c0580fb2068b3754858c9895fc66ea1517b1f87a3ff5328acd76943eae5b12161f469efce500a5343cecd1dc332f2cc9d9b3591c465564c48c28e04ea6ab0350004fe0cceadf86e651727cb1557af92c3bbd719e98f8fb1556dd310e188ee24a12e5727339a9b05e33d767dfb86bb2dc216a7a1d402738181fcc589bf2b2eb04881000888d5fd62cc0f1fe746492d032fa06564f7a29ad78aeaef9271829ebd076ce7347e6872d6f29165968e7a18b57045948de039e3b45efb4ed5dfb64345d2343ea2000ab183649f684ce6d5c1a170403f6f7da7c1a78ea488fe782307bb31ce79a9062056208a3369d346f236e828f87c4a4a0e83059ec162440da143c8c491a6f6a889010c21c47d39a4d342ebde45e95fa9ecf897cf8ee9d24375f40b660796257f7acc5e769cd6e523b4e6792a23a50411bedeebef486089f9fb919011386dfdd013042e010d196ac284de71c5f8b9c2eceadfaf0ee9b456b5f08c3491946e23829ce2ff66df3bab41b1ffab1725f2cd4f6d9229e8cc7494761eb0cb6f8e6562b8fdbd868104000ed421fbea6198271b7bb0b9bd24ecc9658a3739d96fb287d95d64d150db09194326d9a2750aff72e7b3efdf2b083f142c021072ec98635e404ea7eb0afe75e79b0010973fdda770acbc4cc2b55b1f0c2e0437beacb7386b12eb9bb29cbc232939b401486eb3277750c922d83ba146d6b926b7771312de936419d77f3566da6962c236001163c29ea10f90f3dc3840ac4513ae79e1001fa4c92b7a86de894b36c27edeabfd70a09e531dd39c22978e2a283ee6e0ce69f849620384ccac0ca44062db78274b0112c3fc3e6f5bb06055d8e6129b3c33ba877a4f22ea105c1e38a05801a3ceca3b7409455e0db85465e73f62bc717c94427982a79d64bc94c0a63014e878dd8884840064afe0980000e8f10015ccceeb29348f71bdd22ffef43a2a19c1f5b5e17c5cca5411529120182672ade500000000000019460001000000000000000000000000000000000000000000000000000000000b80f2409258181f5ceac8dbffb7030890243caed69a9599d2886d957a9cb7656af3bdb300150000000000000000000000009631288f4050f7cfbf77b77f8540decf6cfc7012000500000000000000000000000000000000000000000000000000000000000000000000000000000000","accessList":null}, error={"code":-32603,"message":"Internal JSON-RPC error.","data":{"code":-32000,"message":"execution reverted"},"stack":"{\n  \"code\": -32603,\n  \"message\": \"Internal JSON-RPC error.\",\n  \"data\": {\n    \"code\": -32000,\n    \"message\": \"execution reverted\"\n  },\n  \"stack\": \"Error: Internal JSON-RPC error.\\n    at new i (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/common-1.js:16:31538)\\n    at s (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/common-1.js:16:34478)\\n    at Object.internal (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/common-1.js:16:35088)\\n    at c (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/background-0.js:7:72700)\\n    at chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/background-0.js:7:73699\\n    at async chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/background-3.js:12:100465\"\n}\n  at new i (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/common-1.js:16:31538)\n  at s (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/common-1.js:16:34478)\n  at Object.internal (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/common-1.js:16:35088)\n  at c (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/background-0.js:7:72700)\n  at chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/background-0.js:7:73699\n  at async chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/background-3.js:12:100465"}, code=CALL_EXCEPTION, version=providers/5.7.2)
    at f.makeError (index.ts:269:28)
    at f.throwError (index.ts:281:20)
    at y (json-rpc-provider.ts:66:16)
    at u.<anonymous> (json-rpc-provider.ts:642:20)
    at Generator.throw (<anonymous>)
    at s (formatter.ts:523:1)
```

Attachment:
![Screenshot 2023-07-13 at 14 43 07](https://github.com/wormhole-foundation/wormhole-connect/assets/10746654/0a60d834-cd32-401a-a278-b0478c0abf7f)

**Comment by AigaD** [comment]
Issue reproduced for transaction WMATIC from Solana (Solflare) to Moonbase (Metamask)
Tested from: https://wormhole-connect.netlify.app/#
Attachment:
<img width="1355" alt="Screenshot 2023-07-18 at 11 30 08" src="https://github.com/wormhole-foundation/wormhole-connect/assets/91047061/0c92c6a7-0f3f-41f7-95e0-38b8d8c115c8">

**Comment by AigaD** [comment]
This issue is reproduced from https://mainnet--wormhole-connect.netlify.app/
After clicking on Claim, transaction fails with error `missing revert data in call exception; Transaction reverted without a reason string`
In attachment transaction from Base -> BSC
Attachment:

https://github.com/wormhole-foundation/wormhole-connect/assets/91047061/3e49d64b-b0b1-4741-8cd9-bfb037ccba4a

**Comment by anondev2323** [comment]
This is actually a few different errors (insufficient amount for gas, JSON RPC error with code 32603, and tx details with no additional error information). They seem to be Metamask related (or Metamask via WalletConnect).  [Here is a guide](https://blog.devgenius.io/tips-to-handle-internal-json-rpc-error-on-metamask-4fc103051a8a) for making sure Metamask is configured correctly, specifically the "Internal JSON RPC 32603 error on MetaMask" section.  Additionally, there's some information which might help us debug

**Comment by aadam-10** [comment]
@anondev2323 @aadam-10 let's check this one and split it into smaller tickets

**Comment by aadam-10** [comment]
Closing this ticket in favor of 3 different ones:

- https://github.com/wormhole-foundation/wormhole-connect/issues/826
- https://github.com/wormhole-foundation/wormhole-connect/issues/827
- https://github.com/wormhole-foundation/wormhole-connect/issues/828

TLDR: deeper in the return message it is written if it was a gas limit issue, the json-rpc issue or really an empty error. Closing this ticket in favor of these 3 as they should be investigated separately.
